server/util/cosmos.js

```diff
@@ -13,176 +18,70 @@ import { PUBSUB_TOPIC_MISC } from '../constant';
 import publisher from './gcloudPub';
 
 import {
-  COSMOS_ENDPOINT as cosmosLCDEndpoint,
+  COSMOS_ENDPOINT as cosmosRPCEndpoint,
   COSMOS_GAS_PRICE,
   COSMOS_DENOM,
-  COSMOS_CHAIN_ID,
 } from '../config/config';
 import { COSMOS_PRIVATE_KEY } from '../config/secret';
 
-const api = axios.create({
-  baseURL: cosmosLCDEndpoint,
-  headers: {
-    'Content-Type': 'application/json',
-  },
-});
-
-function createSigner(privateKey) {
-  const publicKey = secp256k1.publicKeyCreate(privateKey, true);
-  const sha256 = createHash('sha256');
-  const ripemd = createHash('ripemd160');
-  sha256.update(publicKey);
-  ripemd.update(sha256.digest());
-  const rawAddr = ripemd.digest();
-  const cosmosAddress = bech32.encode('cosmos', bech32.toWords(rawAddr));
-  const sign = (msg) => {
-    const msgSha256 = createHash('sha256');
-    msgSha256.update(msg);
-    const msgHash = msgSha256.digest();
-    const { signature } = secp256k1.sign(msgHash, privateKey);
-    return { signature, publicKey };
-  };
-  return { cosmosAddress, sign };
-}
-const cosmosSigner = createSigner(COSMOS_PRIVATE_KEY);
-
-function LIKEToNanolike(value) {
-  return (new BigNumber(value)).multipliedBy(1e9).toFixed();
-}
+let wallet;
+let delegatorAddress;
+let queryClient;
+let signingClient;
 
-export function LIKEToAmount(value) {
-  return { denom: COSMOS_DENOM, amount: LIKEToNanolike(value) };
-}
-export function amountToLIKE(likecoin) {
-  if (likecoin.denom === 'nanolike') {
-    return (new BigNumber(likecoin.amount)).dividedBy(1e9).toFixed();
-  }
-  // eslint-disable-next-line no-console
-  console.error(`${likecoin.denom} is not supported denom`);
-  return -1;
-}
+(() => async function createSigner() {
+  wallet = await DirectSecp256k1Wallet.fromKey(COSMOS_PRIVATE_KEY);
+  const [firstAccount] = await wallet.getAccounts();
+  delegatorAddress = firstAccount.address;
+  const tendermint34Client = await Tendermint34Client.connect(cosmosRPCEndpoint);
+  queryClient = QueryClient.withExtensions(
+    tendermint34Client,
+    setupAuthExtension,
+    setupBankExtension,
+  );
+  signingClient = await SigningStargateClient.connectWithSigner(cosmosRPCEndpoint, wallet);
```

Comment:
I think a better way to write this, is to use async getter that lazy init queryClient/signingClient when they are used
We do not really know when does this async function complete if we just run an arrow function here